### ISSUE TITLE
Direct function to receive Images or Tensors instead of image_paths.

### ISSUE THREAD

**Issue by martian422** [MizzenAI/HPSv3#7]
HPSv3 is a great work, and the community hopes to integrate it into RL training frameworks. However, the default reward() setting only receives img_paths, which may make it a bit complex to evaluate the model's generated contents without saving locally. 

Can the authors add a direct support (like a short cut) to image inputs (like PIL.Images type)? Thanks a lot.

**Comment by martian422** [comment]
Thanks for your kind help!

**Comment by garychan22** [comment]
do we need to resize the image manually? I inputted three 3k * 4k images and the OOM is raised

**Comment by xilanhua12138** [comment]
> do we need to resize the image manually? I inputted three 3k * 4k images and the OOM is raised, but the examples are ok. the gpu is 80G ram.

You might need to check whether 'smart_resize' is working in L139 of hpsv3/inference.py in 'fetch_image'

**Comment by garychan22** [comment]
> > do we need to resize the image manually? I inputted three 3k * 4k images and the OOM is raised, but the examples are ok. the gpu is 80G ram.
> 
> You might need to check whether 'smart_resize' is working in L139 of hpsv3/inference.py in 'fetch_image'

thanks.  the reason is grad accumulation when ``torch.no_grad()`` is not included in the example code.

**Comment by Guanzhou-Ke** [comment]
> > > do we need to resize the image manually? I inputted three 3k * 4k images and the OOM is raised, but the examples are ok. the gpu is 80G ram.
> > 
> > 
> > You might need to check whether 'smart_resize' is working in L139 of hpsv3/inference.py in 'fetch_image'
> 
> thanks. the reason is grad accumulation when `torch.no_grad()` is not included in the example code.

Yep, the original `reward` function will compute the grads during inference. But I have submitted a PR to fix it.
